firedrake/variational_solver.py

```diff
@@ -178,6 +178,7 @@ def update_diffusivity(current_solution):
         """
         assert isinstance(problem, NonlinearVariationalProblem)
 
+        solver_parameters = flatten_parameters(solver_parameters or {})
```

Comment:
Probably needed to do this for `LinearSolver` too?

Reply:
https://github.com/firedrakeproject/firedrake/pull/2624